src/command_modules/azure-cli-search/azure/cli/command_modules/search/tests/latest/test_querykeys.py

```diff
@@ -0,0 +1,66 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.testsdk import ScenarioTest, ResourceGroupPreparer
+import unittest
+
+
+class AzureSearchQueryKeysTests(ScenarioTest):
+
+    @ResourceGroupPreparer(name_prefix='azure_search_cli_test')
+    def test_querykeys_create_delete_list(self, resource_group):
+        self.kwargs.update({
+            'sku_name': 'standard',
+            'name': self.create_random_name(prefix='test', length=24),
+            'key_name_1': self.create_random_name(prefix='test', length=24),
+            'key_name_2': self.create_random_name(prefix='test', length=24),
+            'replica_count': 1,
+            'partition_count': 1,
+        })
+
+        self.cmd('az search services create -n {name} -g {rg} --sku {sku_name}',
+                 checks=[self.check('name', '{name}'),
+                         self.check('sku.name', '{sku_name}'),
+                         self.check('replicaCount', '{replica_count}'),
+                         self.check('partitionCount', '{partition_count}')])
+
+        _querykeys = self.cmd('az search querykeys list -n {name} -g {rg}').get_output_in_json()
+        self.assertTrue(len(_querykeys) == 1)
+        self.assertTrue(_querykeys[0]['name'] is None)
+        self.assertTrue(len(_querykeys[0]['key']) > 1)
+
+        _querykey_1 = self.cmd('az search querykeys create -n {name} -g {rg} --key-name {key_name_1}',
```

Comment:
Aliasing is wrong here. `--key-name` should be aliased to `--name/-n` and what you currently have as `--name` should be aliased to `--search-service-name`

Reply:
So for adminkeys commands, I should also use --search-service-name instead of --name?

Reply:
Correct.